tests/integration/dataframe/test_dataframe.py

```diff
@@ -340,10 +341,134 @@ def test__write_invalid_data__raises(
     ):
         id = test_tables[0]
 
-        frame = models.DataFrame(
+        frame = DataFrame(
             columns=["index", "non_existent_column"],
             data=[["1", "2"], ["2", "2"], ["3", "3"]],
         )
 
         with pytest.raises(ApiException, match="400 Bad Request"):
-            client.append_table_data(id, models.AppendTableDataRequest(frame=frame))
+            client.append_table_data(id, AppendTableDataRequest(frame=frame))
+
+    def test__query_table_data__sorts(self, client: DataFrameClient, create_table):
+        id = create_table(
+            CreateTableRequest(
+                columns=[
+                    int_index_column,
+                    Column(name="col1", data_type=DataType.Float64),
+                    Column(name="col2", data_type=DataType.Float64),
+                ]
+            )
+        )
+
+        frame = DataFrame(
+            data=[["1", "1.5", "5.5"], ["2", "2.5", "6.5"], ["3", "2.5", "7.5"]],
+        )
+
+        client.append_table_data(
+            id, AppendTableDataRequest(frame=frame, end_of_data=True)
+        )
+
+        # TODO: Remove mock when service supports flushing
+        with responses.RequestsMock() as rsps:
+            rsps.add(
+                responses.POST,
+                f"{client.session.base_url}tables/{id}/query-data",
```

Comment:
Are we able to verify the request JSON being sent? These tests aren't actually testing the parameters being passed to the API, just that they can deserialize the response JSON.

Reply:
Turns out the `responses` library makes it really easy to match JSON bodies 🥳